src/player.cpp

```diff
@@ -4705,3 +4705,21 @@ void Player::updateRegeneration()
 		condition->setParam(CONDITION_PARAM_MANATICKS, vocation->getManaGainTicks() * 1000);
 	}
 }
+
+bool Player::isEquippedItem(const Item* item) const
```

Comment:
For this function, TFS already has a way of checking if an item is one of those slots, Player::getInventoryItem. Utilize that method for this one.